en/lessons/advanced/nerves.md

```diff
@@ -0,0 +1,262 @@
+
+Note that you need to use your local network parameters here - in my network there is an unallocated IP`192.168.88.2`, which I am going to use. However, in your case, it may differ.
+
+After adding this, let's try burning and running the changed version of the firmware via `mix firmware.burn`.
+
+If you did as I and started the `ping 192.168.88.2` command before powering the device up, you will be able to see something like this:
+
+```
+Request timeout for icmp_seq 206
+Request timeout for icmp_seq 207
+64 bytes from 192.168.88.2: icmp_seq=208 ttl=64 time=2.247 ms
+64 bytes from 192.168.88.2: icmp_seq=209 ttl=64 time=2.658 ms
+```
+
+This output means that the device now is reachable from the network!
+
+Seems like the best time to launch a new crowdfunded hardware startup, isn't it?
+
+## Network firmware burning
+
+Previously, we were burning SD cards and then physically pushed them into hardware. While it is okay for starters, it's more straightforward to have a network update. That's what `nerves_firmware_ssh` does. It is already present in your project by default and configured to auto-detect and find ssh keys in your folder.
+
+But to use network firmware update, you need to generate an upload script via  `mix firmware.gen.script`. This command will generate new `upload.sh` script for updating the firmware.
+
+If the network has been functional after the previous step, that's it!
+
+You can finally stop switching SD cards: `mix firmware && ./upload.sh 192.168.88.2` is a combination of two commands: first creates a firmware, and the second one burns it via the network and reboots the device.
+
+_Hint: `ssh 192.168.88.2` gives you a proper IEx shell on the device in the context of the app. _
+
+_Troubleshooting: If you don't have an existing ssh key in your home folder, you will have an error `No SSH public keys found in ~/.ssh.`. In this case, you will need to run `ssh-keygen` and re-burn the firmware to use the network update feature._
+
+## Setting up the LED control
+
+To interact with LEDs, you need [nerves_leds](https://github.com/nerves-project/nerves_leds) package installed which, as you know, is done by adding `{:nerves_leds, "~> 0.8", targets: @all_targets},` to `mix.exs` file.
+
+After setting up the dependency, you need to configure the LED list for the given device. For example, for all Raspberry Pi models, there is only one LED onboard: `led0`. Let's use it by adding a `config :nerves_leds, names: [green: "led0"]` line to the `config/config.exs`.
+
+For other devices, you can take a look at the [corresponding part of the nerves_examples project](https://github.com/nerves-project/nerves_examples/tree/master/hello_leds/config).
+
+After configuring the LED itself, we surely need to control it somehow. To do that, we will add a GenServer in `lib/network_led/blinker.ex` with these contents:
```

Comment:
May be worth pointing to the [GenServer](https://elixirschool.com/en/lessons/advanced/otp-concurrency/#genserver) section of Elixir School site. Elixir devs are going to hopefully know what a GenServer is, but those from an embedded development background who are new to Elixir may not.